anchor/subnet_tracker/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+use alloy::primitives::keccak256;
+use alloy::primitives::ruint::aliases::U256;
+use database::{NetworkState, UniqueIndex};
+use log::warn;
+use serde::{Deserialize, Serialize};
+use ssv_types::Cluster;
+use std::ops::Deref;
+use std::time::Duration;
+use task_executor::TaskExecutor;
+use tokio::sync::{mpsc, watch};
+use tokio::time::sleep;
+use tracing::debug;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct SubnetId(#[serde(with = "serde_utils::quoted_u64")] u64);
+
+impl SubnetId {
+    pub fn new(id: u64) -> Self {
+        id.into()
+    }
+}
+
+impl From<u64> for SubnetId {
+    fn from(x: u64) -> Self {
+        Self(x)
+    }
+}
+
+impl Deref for SubnetId {
+    type Target = u64;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+pub enum SubnetEvent {
+    Join(SubnetId),
+    Leave(SubnetId),
+}
+
+pub struct SubnetTracker {
+    events: mpsc::Receiver<SubnetEvent>,
+}
+
+impl SubnetTracker {
+    pub async fn recv(&mut self) -> Option<SubnetEvent> {
+        self.events.recv().await
+    }
+}
+
+pub fn start_subnet_tracker(
+    db: watch::Receiver<NetworkState>,
+    subnet_count: usize,
+    executor: &TaskExecutor,
+) -> SubnetTracker {
+    // a channel capacity of 1 is fine - the subnet_tracker does not do anything else, it can wait.
+    let (tx, rx) = mpsc::channel(1);
+    executor.spawn(subnet_tracker(tx, db, subnet_count), "subnet_tracker");
+    SubnetTracker { events: rx }
```

Comment:
What is `subnet_tracker` doing? It feels a bit confusing to me, it's being called before the `SubnetTracker`instance is created right after.